clippy_lints/src/utils/hir_utils.rs

```diff
@@ -574,4 +581,98 @@ impl<'a, 'tcx> SpanlessHash<'a, 'tcx> {
             },
         }
     }
+
+    pub fn hash_lifetime(&mut self, lifetime: &Lifetime) {
+        if let LifetimeName::Param(ref name) = lifetime.name {
+            match name {
+                ParamName::Plain(ref ident) => {
+                    ident.name.hash(&mut self.s);
+                },
+                ParamName::Fresh(ref size) => {
+                    size.hash(&mut self.s);
+                },
+                _ => {},
+            }
+        }
+    }
+
+    pub fn hash_ty(&mut self, ty: &Ty) {
```

Comment:
I think you should also hash the `hir_id` and the `span` of the `hir::Ty`, or is this intended?

Reply:
So the hashing was meant to be spanless, if I hash the span then would I not find that `&'a T: Copy, &'a T: Clone` have different type hashes?

Reply:
Oh yeah, my bad. never mind then. The discriminant hashing should be added to the `hash_lifetime` function though.